Marlin/src/HAL/STM32/HAL.cpp

```diff
@@ -20,7 +20,7 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  *
  */
-#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC)
+#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC) && !defined(MAPLE_STM32F1)
```

Comment:
If this HAL doesn't need to be compiled in Arduino IDE then we must change how we are bracketing this HAL. I suggest adding `-DHAL_STM32` to the build flags.

Reply:
I was commenting that we should use the macros already defined in the base frameworks... and I took a fast look... it's a mess...

I agree with you that we can create our own macro to enable/disable our hal... Easier and less confusing than theses lots of different but almost identical framework macros...

Reply:
The `STM32GENERIC` flag is set by [danieleff/STM32GENERIC](https://github.com/danieleff/STM32GENERIC) which was originally created to fill in the gaps in Arduino IDE at a time when STM32 support was lacking. Maybe now it is getting obsolete. Or, maybe it's so good now that we should make a HAL for it.